apollo-router/src/spec/field_type.rs

```diff
@@ -80,114 +82,83 @@ impl std::fmt::Display for FieldType {
 }
 
 fn validate_input_value(
```

Comment:
Validation should definitely live in `apollo-rs`. I take it this is execution-specific validation? (i.e. we can't quite strip this out of the router)

Reply:
This is what the spec calls [CoerceVariableValues](https://spec.graphql.org/October2021/#CoerceVariableValues()) except instead of returning `Result<CoercedValues, RequestError>` this version only returns a boolean.

So far apollo-compiler doesn’t deal with variable values at all, but i agree it makes sense to add this to the project scope. Introspection needs it, and https://github.com/apollographql/apollo-rs/pull/645 includes it. I commented in https://github.com/apollographql/apollo-rs/pull/645#issuecomment-1772432060.